src/main/java/seedu/duke/commands/GpaCommand.java

```diff
@@ -0,0 +1,55 @@
+package seedu.duke.commands;
+
+import seedu.duke.exceptions.ModHappyException;
+import seedu.duke.exceptions.ModuleListEmptyException;
+import seedu.duke.tasks.Module;
+import seedu.duke.tasks.ModuleList;
+import seedu.duke.util.Configuration;
+import seedu.duke.util.Grades;
+import seedu.duke.util.StringConstants;
+
+import java.util.Objects;
+
+public class GpaCommand extends Command {
+
+    private static final String GPA_MESSAGE = StringConstants.GPA_MESSAGE;
+
+    private String result;
+    private int mc;
+    private double modularGradePoint;
+    private Grades modularGrade;
+    private float gpa = 0;
+    private int totalMc = 0;
+    private double sumOfProduct = 0.0; // product of modular grade point and mc
+
+    public void calculateGpa(ModuleList moduleList) throws ModHappyException {
+        if (Objects.isNull(moduleList.list)) {
```

Comment:
i think this could be replaced with moduleList.getModuleList() so we do not have to make the list variable public, or have two methods of access

Reply:
yup correct, made changes regarding this